### PR TITLE
*: check if ID token has expired.

### PR DESCRIPTION
**Pull request by rithujohn191**
Addresses #123.

Also added a unit test to verify_test.go

@sym3tri

### CODE REVIEW

File: verify.go [hunk 1]
```diff
@@ -145,6 +145,13 @@ func (v *IDTokenVerifier) Verify(ctx context.Context, rawIDToken string) (*IDTok
 		}
 	}
 
+	// If a checkExpiry is specified, make sure token is not expired.
+	if v.config.checkExpiry != nil {
```
Comment (kans) [inline verify.go]: Would this ever be anything other than time.Now?

Comment (squat) [inline verify.go]: you might want to preemptively check if it will expire next week

Comment (rithujohn191) [inline verify.go]: Looks like VerifyExpiry() always sets "checkExpiry = time.Now"

Comment (rithujohn191) [inline verify.go]: @squat Verifying token expiry is optional, it is implemented only if checkExpiry is set by VerifyExpiry().

**Pull request merged**